app/forms/tasks.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Tasks
+  class << self
+    def form_for(slug)
+      FORM_HANDLERS.fetch(slug, BaseForm)
+    end
+  end
+
+  FORM_HANDLERS = {
+    "check-breach-report" => BaseForm,
+    "investigate-and-decide" => BaseForm,
+    "review-recommendation" => BaseForm,
+    "serve-notice-and-monitor-compliance" => BaseForm,
+    "process-an-appeal" => BaseForm,
+
+    "check-report-details" => CheckReportDetailsForm
+  }.freeze
```

Comment:
This feels like it should be defined in the workflow yaml with a method override on the task to return the default BaseForm. Also we need to handle the nesting of tasks as this is just a flat list.